packages/wallet/backend/src/auth/service.ts

```diff
@@ -54,21 +57,44 @@ export class AuthService implements IAuthService {
     return user
   }
 
+  public async resendVerifyEmail({
+    email
+  }: resendVerifyEmailArgs): Promise<void> {
+    const user = await this.userService.getByEmail(email)
+
+    // TODO: Prevent timing attacks
+    if (!user) {
+      throw new Unauthorized('Invalid account')
+    }
+    const token = getRandomToken()
+    await this.userService.resetVerifyEmailToken({
+      email,
+      verifyEmailToken: hashToken(token)
+    })
+
+    await this.emailService.sendVerifyEmail(email, token).catch((e) => {
+      this.logger.error(
+        `Error on sending verify email for user ${user.email}`,
+        e
+      )
+    })
+  }
+
   public async authorize(args: AuthorizeArgs): Promise<AuthorizeResult> {
     const user = await this.userService.getByEmail(args.email)
 
     // TODO: Prevent timing attacks
     if (!user) {
-      throw new Unauthorized('Invalid credentials')
+      throw new Unauthorized('Invalid credentials.')
     }
 
-    const isValid = await user.verifyPassword(args.password)
-    if (!isValid) {
-      throw new Unauthorized('Invalid credentials')
+    if (!user.isEmailVerified) {
```

Comment:
verifying the password first is the correct approach.
Telling the user that the email is not verified even if the password is incorrect increases security risks.